plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEnginePublishOperationTest.java

```diff
@@ -66,10 +66,13 @@ public void setUp() throws IOException, CoreException, InterruptedException {
     serverProject = projects.stream().filter(isServerProject).findFirst().get();
     sharedProject = projects.stream().filter(isSharedProject).findFirst().get();
 
+    // https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1798
     for (int i = 0; i < 100; i++) {
       serverModule = ServerUtil.getModule(serverProject);
       sharedModule = ServerUtil.getModule(sharedProject);
-      if (serverModule != null && sharedModule != null) {
+      if (serverModule != null && sharedModule != null
+          && serverProject.getFile("bin/sox/server/GreetingServiceImpl.class").exists()
+          && sharedProject.getFile("bin/sox/shared/GreetingService.class").exists()) {
         break;
       }
       Thread.sleep(100);
```

Comment:
Could you put in a call to `ThreadDumpingWatchdog` here so that we can see the pending jobs?

Reply:
Added `ThreadDumpingWatchdog.report("Until modules are fully ready", stopwatch)`.

The only possible problem could be that we may never take our time to look for a report if this PR really fixes #1798 and tests always succeed.